backend/src/api/blocks.ts

```diff
@@ -341,9 +345,33 @@ class Blocks {
       for (let y = 0; y < regexes.length; ++y) {
         const regex = new RegExp(regexes[y], 'i');
         const match = asciiScriptSig.match(regex);
-        if (match !== null) {
-          return pools[i];
+        if (match !== null && !matchingPoolsSlugs[pools[i].slug]) {
+          matchingPools.push(pools[i]);
+          matchingPoolsSlugs[pools[i].slug] = true;
+        }
+      }
+    }
+
+    if (matchingPools.length === 1) {
+      return matchingPools[0];
+    }
+    
+    // Conflicts - Resolve manually for now
+    if (matchingPools.length > 1) {
+      const pick = (matchingPools: PoolTag[], slug: string): PoolTag => {
+        for (const pool of matchingPools) {
+          if (pool.slug === slug) {
+            return pool;
+          }
         }
+        return matchingPools[0];
+      };
+      if (id === '00000000000000000003aff64602a1687539515d9e2c44dc1f8c31289901781a') {
```

Comment:
It's just wrong to hard code a single block in the code like this.

If we are to do exceptions in the matching we need a better way.

Reply:
What is your suggested better way?

We could move the hardcoded block ids into a dedicated file to avoid spamming the already messy `blocks.ts` file. But in terms of "hardcoding" I don't really see a way around it in the example I've provided https://mempool.space/block/0000000000000000001b9460202a7acc512630988f8204df39e5c7d605587425.

An alternative approach would be to have some UX elements to show that the block cannot be assigned to a miner with certainty and simply exclude it from the stats.

Reply:
Aren't these very related to the whole pools json database matching. So it should be in that file or a separate file that share the same commit history. Everyone using the mining-pools repo should get the exceptions list, and it can be collaborative.

Reply:
That's a good suggestion. @0xB10C what do you think of this approach?

Reply:
I think having an exception list that we manually define is OK. It sounds like a lot of work though.

One approach I've been happy with is:

1. try identifying by coinbase output address first
2. if unsuccessful, try tagging by script tag 
3. if unsuccessful, miner is unknown

The coinbase output address is a better metric to check first as it's quite expensive to game. As a mining pool you can include arbitrary tag in the script at close to zero cost. By putting another pool's address in the output, you'd 
essentially gift the coins to another pool.